draft-ietf-core-yang-cbor.md

```diff
@@ -261,6 +261,22 @@ by the environment.
 
 Mechanisms and processes used to assign SIDs to YANG items and to guarantee their uniqueness are outside the scope of the present specification.
 If SIDs are to be used, the present specification is used in conjunction with a specification defining this management.
+The objectives for such a management scheme need to include:
+
+* Keep the mapping from YANG SIDs to YANG names ({{name}}) a mathematical
+  function — a single YANG SID is only ever associated with a
+  single YANG name.
```

Comment:
A "path name" is not clear and it is not used in the YANG module.
Does this mean the mapping between the [namespace, identifier] tuple to a specific SID

This is only relevant for draft=false.
For a draft SID file, this property only applies to the SIDs inherited from the  parent-stable SID file (if any)
For a published SID file, this property applies to all SIDs

Reply:
The status discussion is over in core-sid; I added a brief preview of that.
"YANG name" refers to Section 3.3, this is called "YANG item identifier" there, but YANG name is specific to the way we do the identifiers in YANG-CBOR.
(I don't see "path name" anywhere.)

Reply:
You mean this text:

                    name = [identifier ":"] identifier

                    if:name
                    mynode

The ABNF for "identifier" allows simple names

             identifier          = (ALPHA / "_")
                                       *(ALPHA / DIGIT / "_" / "-" / ".")

YANG only requires QNames to be unique within the sibling set
because it is hierarchical by design.

Reply:
OK, my brain needs a serious terminology overhaul.
I was thinking about what is called "identifier" in a SID-File, which in namespace "data" is a schema-node-path starting with a qualified name, while it is a simple (unqualified!) yang:yang-identifier for "feature", "identity", "module".
So the mapping would be between a SID on one side and a schema-node-path or a qualified name built from module name and unqualified identifier (depending on namespace).

Reply:
The text about 7950 ABNF for "identifier" need to change so it supports the typedef used for identifier leaf
Right now it just accounts for yang-identifier, not schema-node-path


        type union {
             type yang:yang-identifier;
             type schema-node-path;
        }

The desired binding is described in the YANG list "item"

         key "namespace identifier";
         unique "sid";

That is what I meant by [namespace identifier] to SID mapping

Reply:
I'm not sure we need to change that ABNF, because a single [id:]id is fine for a map key.
What we are after here is a concept of the thing SIDs map to.
To me, that seems to be [module-name, "data"/"feature"/"identity"/"module", schema-node-path/identifier].
module-name because unfortunately the non-data items don't use the full qualified name in the SID file.
Or, really, in CDDL:

[module-name, ("data", schema-node-path) // ("feature"/"identity"/"module", identifier)]